packages/react-pose/src/components/Transition/children.ts

```diff
@@ -62,31 +57,36 @@ const mergeChildren = ({
   );
 
   incomingChildren.forEach(child => {
-    const newChildProps = {
-      ...propsForChild,
-      ...filterChildProps(child.props)
-    };
+    const newChildProps = {};
 
     if (entering.has(child.key as string)) {
+      // If child is entering
       newChildProps.initialPose = preEnterPose;
-      newChildProps._pose = enterPose;
+      newChildProps._pose = enterPose; // TODO: Remove _pose and merge with child.props.pose
     } else if (moving.has(child.key as string) && flipMove) {
+      // If child is moving and we're using `flip`
       newChildProps._pose = [enterPose, 'flip'];
     } else {
+      // If child is moving
       newChildProps._pose = enterPose;
     }
 
-    children.push(cloneElement(child, newChildProps));
+    mergedChildren.push(cloneElement(child, newChildProps));
   });
 
   leaving.forEach(key => {
-    const child = displayedChildren.find(c => c.key === key);
+    const child = children.find(c => c.key === key);
+    const removeChildFromTree = removeFromTree(key)
+
     const newChild = cloneElement(child, {
       _pose: exitPose,
-      onPoseComplete: removeFromTree(key),
+      onPoseComplete: pose => {
+        removeChildFromTree()
+        const { onPoseComplete } = child.props
+        // TODO: call original onPoseComplete prop also in react-pose-core, vue-pose
```

Comment:
probably should be done as part of this PR - but didn't want to touch those without your approval first

Reply:
Yeah great idea